ang/civicase/CaseListTable.js

```diff
@@ -210,13 +297,31 @@
         });
     }
 
+    /**
+     * Initialise watchers
+     */
     function initiateWatchers () {
       $rootScope.$on('civicase::fetchMoreContactsInformation', fetchContactsData);
       $scope.$watchCollection('sort', updateCases);
       $scope.$watchCollection('page', updateCases);
-      $scope.$watch('cases', function (cases) {
-        $scope.selectedCases = _.filter(cases, 'selected');
-      }, true);
+      $scope.$watch('cases', casesWatcher, true);
+    }
+
+    /**
+     * Case Watcher - Updates the checkbox if a case is selected
+     *
+     * @params {array} cases
+     */
+    function casesWatcher (cases) {
+      // if case is in selectedCases array update the UI model (checkbox)
+      _.each(cases, function (item, index) {
```

Comment:
I think we discussed that we will be storing the IDs only. can we do that?

Reply:
I think that we discussed it already that  we will be storing the full case object. As if we just store the keys, then we have to create new services for fetching the details of each case with an id and it will be a complex architecture and might introduce regressions.